### DISCUSSION TITLE
Specify selected / highlighted cells programatically

### DISCUSSION THREAD

**Discussion by sazzamac** [iddan/react-spreadsheet#292] [Ideas]
I'd like to be able to programatically assign highlighted (selected) cells without clicking and dragging.

For example by passing a set of point indices to the spreadsheet to highlight.

Is this possible already? If so, how can this be done?

**Comment by iddan** [comment]
This is not currently possible, but PRs are welcome! Currently, I don't actively develop new features myself but I would love to review a PR.

**Comment by JPBotelho** [comment]
I was able to implement a workaround by looking for the cell elements in the DOM and triggering their mousedown events. 

Example:

`const cell = document.querySelectorAll('td.Spreadsheet__cell')[1];

const event = new MouseEvent('mousedown', {

  view: window,

  bubbles: true,

  cancelable: true

});

cell.dispatchEvent(event);
`

The tr elements also have their row number, so you con probably search for that and then find the cell you want. 

This is however limited to highlighting a single cell
